docs/advanced/analytics/plugins.mdx

```diff
@@ -0,0 +1,340 @@
+  means that you can try to duplicate the scripts content and adapt it to your
+  integration.
+- If none of the solutions above work, you can always try to load the script
+  several times by adding a `?random=${new Date().getTime()}` at the end of the
+  URL. This will trick the browser into thinking they are different scripts and
+  allow you to load it multiple times.
+
+Implementing a great tagging plan for an e-commerce application is a tough
+journey. If you have any further questions about how to implement them in
+Front-Commerce, please <ContactLink />. We'll be happy to answer them.
+
+## Common Plugins
+
+Here is a list of frequently used plugins across e-commerce shops
+
+### Google Analytics 4
+
+Install the
+[google-analytics plugin](https://getanalytics.io/plugins/google-analytics/)
+
+```shell
+npm install @analytics/google-analytics
+```
+
+Configuration example in `src/config/analytics.js`
+
+```js title="src/config/analytics.js"
+{
+  name: "google-analytics",
+  needConsent: true,
+  settings: (authorization) => {
+    return {
+        measurementIds: ['G-abc123'],
+        gtagConfig:{
+            anonymize_ip: !authorization
+        }
+      },
+  },
+  script: () => import("@analytics/google-analytics"),
+}
+```
+
+### Google Tag Manager
+
+Install the
+[google-tag-manager plugin](https://getanalytics.io/plugins/google-tag-manager/)
+
+```shell
+npm install @analytics/google-tag-manager
+```
+
+Configuration example in `src/config/analytics.js`
+
+```js title="src/config/analytics.js"
+{
+  name: "google-tag-manager",
+  needConsent: true,
+  settings: (authorization, otherAuthorizations) => {
+    containerId: "GTM-123xyz",
+    // the userConsents option is a specific key that the plugin will use and expose in the GTM dataLayer
+    userConsents: otherAuthorizations
+  }
+  script: () => import("@segment/analytics.js-integration-google-tag-manager"),
+}
+```
+
+Update your <abbr title="Content Security Policy">CSP</abbr> in
+`src/config/website.js` according to the tags you use (see
+[Google Tag Manager's documentation](https://developers.google.com/tag-platform/tag-manager/web/csp)
+for more details)
+
+In GTM, you will then be able to leverage several specific things configured in
+your plugins.
+
+First, the `userConsents` configuration option will be pushed to your dataLayer
+as the `userConsents` value. You can reference it from a Variable in GTM. Here
+is an example:
+
+<Figure>
+
+![Screenshot of a GTM Variable configured to expose the user consents](./assets/gtm-datalayer-variable.png)
+
+</Figure>
+
+Then, you can leverage the `UserConsentUpdated` event tracked whenever users
+update their consent preferences. You could create triggers to enable scripts to
+load / remove (depending on the `userConsents` value). Here is an example:
+
+<Figure>
+
+![Screenshot of a GTM Trigger configured to detect when users gave their consent to a specific integration](./assets/gtm-trigger-example.png)
+
+</Figure>
```

Comment:
I need to verify this behavior as I have not yet understood what/how the datalayer worked before, so requires some investigation, 
which is attached to a window object and retrieved via the [dataLayerName](https://getanalytics.io/plugins/google-tag-manager/#configuration-options-for-browser) config key for the analytics.io plugin